### PR TITLE
Session modal (via route)

### PR DESCRIPTION
**Pull request by Illizian**

### CODE REVIEW

File: app/lib/schedule.ts [hunk 1]
```diff
@@ -12,3 +12,13 @@ export const deserializeSchedule = (
...
+
+export const getSessionById = (schedule: ScheduleData, id: string) => {
+  return schedule
+    .reduce(
+      (sessions: Array<Session>, day) => [...day.sessions, ...sessions],
+      []
+    )
+    .reduce((talks: Array<Talk>, sessions) => [...sessions.talks, ...talks], [])
+    .find((talk) => talk.id.toString() === id);
+};
```
Comment (tom-sherman) [inline app/lib/schedule.ts]:
```suggestion
export const getSessionById = (schedule: ScheduleData, id: Talk["id"]) =>
  schedule
    .flatMap((day) => day.sessions.flatMap((session) => session.talks))
    .find((talk) => talk.id === id);
```

Using `flatMap` is easier to read IMO

Accepting the `Talk["id"]` does two things: first it means the function is correctly coupled to the Talk domain type. Secondly it's more explicit about what the caller needs to do and is more type safe.

File: app/lib/airtable.server.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ const airtableResponseSchema = <TSchema extends AnySchema>(
     records: z.array(recordSchema),
   });
 
-const stringArraySchema = z.array(z.string());
+const stringArraySchema = z.array(z.string().nullable());
```
Comment (tom-sherman) [inline app/lib/airtable.server.ts:60]: Why would Airtable be sending nulls in the array? I haven't observed this to be the case when interacting with the API, I think the nullable can be removed.

File: app/lib/airtable.server.ts [hunk 3]
```diff
@@ -75,6 +75,10 @@ const scheduleResponseSchema = airtableResponseSchema(
       id: z.number(),
       Start: z.string(),
       Title: z.optional(z.string()),
+      Website: z.optional(stringArraySchema).default([]),
+      LinkedIn: z.optional(stringArraySchema).default([]),
+      Twitter: z.optional(stringArraySchema).default([]),
+      AvatarUrl: z.optional(stringArraySchema).default([]),
```
Comment (tom-sherman) [inline app/lib/airtable.server.ts:81]: Has the data model/API changed? When I hit it these are optional strings, not string arrays.

File: app/lib/airtable.server.ts (hunk 2 above)
Comment (Illizian) [reply app/lib/airtable.server.ts:60]: It does for calculated fields on related columns
image

File: app/lib/airtable.server.ts (hunk 3 above)
Comment (Illizian) [reply app/lib/airtable.server.ts:81]: As with previous note

**Pull request merged**